crates/swift-integration-tests/src/async_function.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(dead_code)]
```

Comment:
I faced the same issue as #270 - tests wouldn't even compile without this, since `-D warnings` is enabled for them. I also had to make this change to `result.rs` below. Maybe there's a better solution, this was just what I did to get tests running

Reply:
Please document:

- why we're allowing dead code
- what needs to happen to remove this allow dead code
- link to the issue

Seems like this can be deleted once #270 is closed. So, yeah just mention that.

This will make it easier for us to eventually stop allowing dead code.

---

Please also apply this feedback to `result.rs`